Source/WebKit/UIProcess/API/haiku/WebViewBase.cpp

```diff
@@ -65,6 +66,21 @@ void WebViewBase::paint(const IntRect& dirtyRect)
 {
 }
 
+void WebViewBase::MessageReceived(BMessage* message)
+{
+    switch (message->what) {
+        case B_MOUSE_DOWN:
```

Comment:
I would prefer that we use the official functions (MouseDown, MouseUp, etc). They can access the message if needed by using the CurrentMessage() function if needed for extra details about the mouse event.

Handling the messages directly does not work in BeOS, and arguably shouldn't work in Haiku either (it breaks some BeOS applications). By intercepting the message here and not forwaring it to BView you are possibly preventing some OS handling from happening.